google/cloud/bigquery/integration_tests/bigquery_read_integration_test.cc

```diff
@@ -80,6 +81,19 @@ std::int64_t CountRowsFromStream(
   return num_rows;
 }
 
+template <typename Functor>
+google::cloud::internal::invoke_result_t<Functor> StatusRetryLoop(
+    Functor&& operation,
+    std::vector<int> const& delay_seconds = {10, 30, 60, 60}) {
+  google::cloud::internal::invoke_result_t<Functor> status;
+  for (auto delay : delay_seconds) {
+    status = operation();
+    if (status.ok()) break;
```

Comment:
Q. Is there some distinction between retryable and non-retryable errors that we should make?  Can we use retry/backoff policies instead.

Reply:
➕1  You could use the existing retry loop with a custom idempotency policy:

```cc
class TestIdempotencyPolicy : public BigQueryReadConnectionIdempotencyPolicy {
 public:
  virtual ~TestIdempotencyPolicy() override = default;

  /// Create a new copy of this object.
  std::unique_ptr<BigQueryReadConnectionIdempotencyPolicy> clone() const override {
    return absl::make_unique<TestIdempotencyPolicy>();
  }

  google::cloud::Idempotency CreateReadSession(
      google::cloud::bigquery::storage::v1::CreateReadSessionRequest const&) override {
    return google::cloud::kIdempotent;
  }

  google::cloud::Idempotency SplitReadStream(
      google::cloud::bigquery::storage::v1::SplitReadStreamRequest const&) override {
    return google::cloud::kIdempotent;
  }
};
```


And initialize the `*Connection` with that policy.